### PR TITLE
Add initial Grades to native SpeedGrader

### PR DESCRIPTION
**Pull request by vanwagonet**
Makes drawerState stick across submissions
Fixes bad centering of no submission text
Adds more grading info to models
Adds a pretty unsupported screen for moderated grading

refs: MBL-14816
affects: Teacher
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/53503741/p/cbe9ee2dcc83c581631878636a7df381">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/53503746/p/f2f11c092227d2dce41d3fda866ae58c">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/53503735/p/573f976e0a3b33fcd69605f455d349bc">
    
  </a>
</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.23% | 93.24% | -0.01%

[MBL-14816](https://instructure.atlassian.net/browse/MBL-14816)

#### Affected Apps: Teacher
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: rn/Teacher/ios/Teacher/SpeedGrader/SpeedGraderViewController.swift [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                    .font(.semibold16).accentColor(Color(Brand.shared.linkColor))
+                    .padding(16)
+                    .identifier("SpeedGrader.emptyCloseButton")
+            }
+            EmptyPanda(.Space,
+                title: Text("No Submissions"),
+                message: Text("It seems there aren't any valid submissions to grade.")
+            )
+        }
+    )
```
Comment (armstrongnate) [inline rn/Teacher/ios/Teacher/SpeedGrader/SpeedGraderViewController.swift:122]: Mixing and matching swiftui and uikit like this is pretty cool.

File: rn/Teacher/ios/Teacher/SpeedGrader/SubmissionGrader.swift [hunk 2]
```diff
@@ -193,16 +213,15 @@ struct SubmissionGrader: View {
...
-                    case .grades:
+            GeometryReader { geometry in
+                HStack(spacing: 0) {
+                    VStack(spacing: 0) {
                         SubmissionGrades(assignment: assignment, submission: submission)
                             .clipped()
                         Spacer().frame(height: bottomInset)
-                    case .comments:
+                    }
+                        .frame(width: geometry.size.width, height: geometry.size.height)
```
Comment (armstrongnate) [inline rn/Teacher/ios/Teacher/SpeedGrader/SubmissionGrader.swift:223]: I'm a SwiftUI noob. Help me understand why we need to set the frame on all these tabs to the size of the geometry reader?

File: rn/Teacher/ios/Teacher/SpeedGrader/SubmissionViewer.swift [hunk 3]
```diff
@@ -28,35 +28,20 @@ struct SubmissionViewer: View {
...
+                WebView(url: url).onLink(handleLink)
             }
         case .media_recording:
             VideoPlayer(url: submission.mediaComment?.url)
         case .online_text_entry:
-            WebView(html: submission.body)
-                .onLink(handleLink)
-                .onChangeSize(handleSize)
-                .highPriorityGesture(DragGesture(), including: hasOverflow ? .all : .subviews)
+            WebView(html: submission.body).onLink(handleLink)
```
Comment (armstrongnate) [inline rn/Teacher/ios/Teacher/SpeedGrader/SubmissionViewer.swift:44]: Well that cleaned up nicely.

**Review by armstrongnate — APPROVED** [review]
One question. Looks good to me. QA +1

File: rn/Teacher/ios/Teacher/SpeedGrader/SubmissionGrader.swift (hunk 2 above)
Comment (vanwagonet) [reply rn/Teacher/ios/Teacher/SpeedGrader/SubmissionGrader.swift:223]: These are all inside an HStack. Without explicit sizing they will all want to all fit on screen at once. We want each to fit in the outer space and get pushed off screen when another tab is selected.

**Review by Dorene72 — APPROVED** [review]

File: rn/Teacher/ios/Teacher/SpeedGrader/SubmissionGrader.swift (hunk 2 above)
Comment (armstrongnate) [reply rn/Teacher/ios/Teacher/SpeedGrader/SubmissionGrader.swift:223]: Okay duh that makes sense. Thanks.

**Review by armstrongnate — APPROVED** [review]
Grades tab is looking good. 👌

**Pull request merged**